lib/cog/commands/pluck.ex

```diff
@@ -0,0 +1,82 @@
+defmodule Cog.Commands.Pluck do
+  use Spanner.GenCommand.Base, bundle: Cog.embedded_bundle, enforcing: false, calling_convention: :all
+  alias Cog.Helpers, as: CogHelpers
+  alias Cog.Command.Helpers, as: CommandHelpers
+
+  @moduledoc """
+  Returns the specified fields from a set of inputs.
+
+  ## Example
+
+      @bot #{Cog.embedded_bundle}:rules --list --for-command="#{Cog.embedded_bundle}:permissions" | #{Cog.embedded_bundle}:pluck --fields=id
+      > { "id": "91edb472-04cf-4bca-ba05-e51b63f26758" }
+        { "id": "738ce922-14cd-4abc-fe94-81658707229c" }
+      @bot #{Cog.embedded_bundle}:rules --list --for-command="#{Cog.embedded_bundle}:permissions" | #{Cog.embedded_bundle}:pluck --fields=rule,command
+      > { "rule": "operable:manage_users",
+          "command": "operable:permissions" }
+        { "rule": "operable:manage_users",
+          "command": "operable:permissions" }
+
+  """
+
+  option "fields", type: "string", required: false
+
+  defstruct req: nil, fields: nil, input: nil, output: nil, errors: []
+
+  def handle_message(req, state) do
+    case req |> validate |> execute |> format do
+      {:ok, data} ->
+        {:reply, req.reply_to, data, state}
+      {:error, error} ->
+        {:error, req.reply_to, error, state}
+    end
+  end
+
+  defp validate(req) do
+    %__MODULE__{req: req}
+    |> validate_inputs
+    |> validate_options
+  end
+
+  defp validate_options(%__MODULE__{req: %{options: %{"fields" => opt_fields}}, input: item}=state) do
+    fields = String.split(opt_fields, ",")
+    case Enum.reduce(fields, true, &contains_field(item, &1, &2)) do
```

Comment:
If I list fields that I'd like to pluck and one or more are missing, I'd like to know that. Or if I've misspelled a field name. Otherwise, it's possible to silently fail.